opentelemetry-api/src/opentelemetry/context/propagation/tracecontexthttptextformat.py

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
+import re
```

Comment:
I'd prefer not using regular expressions for this, as the format should be simple enough to process with `"".split("-")`. Also, I think implementing https://www.w3.org/TR/trace-context/#versioning-of-traceparent would be easier that way. But that is no strong opinion.

Reply:
that would require checking the validity of the individual values, but I can take a look. regex is very efficient for these type of operations.

I was just curious: the spec looks like it provides no guarantees around the format of future traces. so the most we could do is split on first "-", check the version, and pass-through or ignore if the version is above 00. would that be the right forward-compatible behavior?

Reply:
I think regex is the most efficient way (scientifically) to handle string parsing - if implemented using the correct [Ken Thompson DFA/NFA](https://en.wikipedia.org/wiki/Thompson%27s_construction). No strong opinion though :)